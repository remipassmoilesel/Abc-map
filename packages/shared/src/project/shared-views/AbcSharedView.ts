/**
 * Copyright © 2023 Rémi Pace.
 * This file is part of Abc-Map.
 *
 * Abc-Map is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of
 * the License, or (at your option) any later version.
 *
 * Abc-Map is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General
 * Public License along with Abc-Map. If not, see <https://www.gnu.org/licenses/>.
 */

import { AbcView } from '../AbcView';
import { AbcTextFrame } from '../text-frame';
import { AbcScale } from '../scale';
import { AbcNorth } from '../north';

export interface AbcSharedView {
  id: string;
  title: string;
  view: AbcView;
  layers: LayerState[];
  textFrames: AbcTextFrame[];
  scale?: AbcScale;
  north?: AbcNorth;
}

// Each view can show or hide layers
export interface LayerState {
  layerId: string;
  visible: boolean;
}
