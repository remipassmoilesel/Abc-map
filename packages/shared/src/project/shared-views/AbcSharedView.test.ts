/**
 * Copyright © 2023 Rémi Pace.
 * This file is part of Abc-Map.
 *
 * Abc-Map is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of
 * the License, or (at your option) any later version.
 *
 * Abc-Map is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General
 * Public License along with Abc-Map. If not, see <https://www.gnu.org/licenses/>.
 */

import { AbcSharedView } from './AbcSharedView';

/**
 * If this test fail, you should write a migration script then adapt it
 */
describe('AbcSharedView', () => {
  it('should not change without migration', () => {
    const current: AbcSharedView = {
      id: 'test-id',
      title: 'test-title',
      view: {
        center: [12, 34],
        resolution: 5678,
        projection: { name: 'test-projection' },
        rotation: 0,
      },
      layers: [{ layerId: 'test-layer-id', visible: true }],
      textFrames: [],
      scale: { x: 80, y: 180 },
    };

    expect(current).toMatchSnapshot();
  });
});
