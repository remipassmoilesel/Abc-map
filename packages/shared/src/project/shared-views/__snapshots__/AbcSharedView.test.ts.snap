// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbcSharedView should not change without migration 1`] = `
Object {
  "id": "test-id",
  "layers": Array [
    Object {
      "layerId": "test-layer-id",
      "visible": true,
    },
  ],
  "scale": Object {
    "x": 80,
    "y": 180,
  },
  "textFrames": Array [],
  "title": "test-title",
  "view": Object {
    "center": Array [
      12,
      34,
    ],
    "projection": Object {
      "name": "test-projection",
    },
    "resolution": 5678,
    "rotation": 0,
  },
}
`;
